src/test/java/org/opensearch/flowframework/bwc/FlowFrameworkBackwardsCompatibilityIT.java

```diff
@@ -0,0 +1,189 @@
+
+    private static final ClusterType CLUSTER_TYPE = ClusterType.parse(System.getProperty("tests.rest.bwcsuite"));
+    private static final String CLUSTER_NAME = System.getProperty("tests.clustername");
+
+    @Override
+    @Before
+    public void setUp() throws Exception {
+        super.setUp();
+
+        setupSettings();
+    }
+
+    private void setupSettings() throws IOException {
+        // Enable Flow Framework Plugin Rest APIs
+        Response response = TestHelpers.makeRequest(
+            client(),
+            "PUT",
+            "_cluster/settings",
+            null,
+            "{\"transient\":{\"plugins.flow_framework.enabled\":true}}",
+            List.of(new BasicHeader(HttpHeaders.USER_AGENT, ""))
+        );
+        assertEquals(200, response.getStatusLine().getStatusCode());
+    }
+
+    @Override
+    protected final boolean preserveIndicesUponCompletion() {
+        return true;
+    }
+
+    @Override
+    protected final boolean preserveReposUponCompletion() {
+        return true;
+    }
+
+    @Override
+    protected boolean preserveTemplatesUponCompletion() {
+        return true;
+    }
+
+    @Override
+    protected final Settings restClientSettings() {
+        return Settings.builder()
+            .put(super.restClientSettings())
+            // increase the timeout here to 90 seconds to handle long waits for a green
+            // cluster health. the waits for green need to be longer than a minute to
+            // account for delayed shards
+            .put(OpenSearchRestTestCase.CLIENT_SOCKET_TIMEOUT, "90s")
+            .build();
+    }
+
+    private enum ClusterType {
+        OLD,
+        MIXED,
+        UPGRADED;
+
+        public static ClusterType parse(String value) {
+            switch (value) {
+                case "old_cluster":
+                    return OLD;
+                case "mixed_cluster":
+                    return MIXED;
+                case "upgraded_cluster":
+                    return UPGRADED;
+                default:
+                    throw new AssertionError("unknown cluster type: " + value);
+            }
+        }
+    }
+
+    @SuppressWarnings("unchecked")
+    public void testBackwardsCompatibility() throws Exception {
+        String uri = getUri();
+        Map<String, Map<String, Object>> responseMap = (Map<String, Map<String, Object>>) getAsMap(uri).get("nodes");
+        for (Map<String, Object> response : responseMap.values()) {
+            List<Map<String, Object>> plugins = (List<Map<String, Object>>) response.get("plugins");
+            Set<Object> pluginNames = plugins.stream().map(map -> map.get("name")).collect(Collectors.toSet());
+            String workflowId = createNoopTemplate();
+            Template t = getTemplate(workflowId);
+            switch (CLUSTER_TYPE) {
+                case OLD:
+                    assertTrue(pluginNames.contains("opensearch-flow-framework"));
```

Comment:
This is common for all the cases. Can be moved above switch?

Reply:
It can actually be removed since it's the same for all cases.  I figure it didn't hurt to leave as a placeholder if we ever end up changing the plugin name like AD did.